solarwinds_apm/trace/base_metrics_processor.py

```diff
@@ -90,9 +90,13 @@ def calculate_span_time(
         self,
         start_time: int,
         end_time: int,
+        time_conversion: int = 1e3,
     ) -> int:
-        """Calculate span time in microseconds (us) using start and end time
-        in nanoseconds (ns). OTel span start/end_time are optional."""
+        """Calculate span time (via time_conversion e.g. 1e3, 1e6)
+        using start and end time in nanoseconds (ns). OTel span
+        start/end_time are optional."""
```

Comment:
Curious why otel span start/end times are optional?

Reply:
They're technically optional for [ReadableSpan](https://github.com/open-telemetry/opentelemetry-python/blob/1061d96562990b1c2b596ace87239e7f9616589a/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L368-L369) in terms of init args. I think auto calculation by the SDK (maybe [this?](https://github.com/open-telemetry/opentelemetry-python/blob/1061d96562990b1c2b596ace87239e7f9616589a/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L880)) is the main use case for either instrumentors or start_as_current_span so I imagine almost always there are start/end times. I don't know when there wouldn't be but might as well catch for that case.

Reply:
Gotcha!